File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.cpp
class Solution {
public:
    int maxVowels(string s, int k) {
        int maxV = 0;
        int curV = 0;
        
        for( int i = 0; i < k; i++) {
            if( s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u')
                curV++;
        }
        
        maxV = curV;
        for( int i = k; i < s.size(); i++) {
            if( s[i] == 'a' || s[i] == 'e' || s[i] == 'i' || s[i] == 'o' || s[i] == 'u')
                curV++;
            if( s[i-k] == 'a' || s[i-k] == 'e' || s[i-k] == 'i' || s[i-k] == 'o' || s[i-k] == 'u')
                curV--;
            
            maxV = max(maxV, curV);
        }
        
        return maxV;
    }
};

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution(object):
    def maxVowels(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        vowelset = set('aeiou')
        curvow = 0
        
        for i in range(k):
            if s[i] in vowelset:
                curvow += 1
                
        maxvow = curvow
        
        for i in range(k, len(s)):
            if s[i] in vowelset:
                curvow += 1
            if s[i-k] in vowelset:
                curvow -= 1
                
            maxvow = max(maxvow, curvow)
        
        return maxvow